File: code wars/7kyu/Count the Digit/main.go
package kata

import (
	"strconv"
	"strings"
)

func NbDig(n int, d int) int {
	count := 0
	for i := 0; i <= n+1; i++ {
		arr := strings.Split(strconv.Itoa(i*i), "")
		for _, v := range arr {
			if v == strconv.Itoa(d) {
				count++
			}
		}
	}
	return count
}

// Can be golfed like this (saw on code wars)

func NbDigTwo(n int, d int) (out int) {
	for i := 0; i <= n; i++ {
		out += strings.Count(strconv.Itoa(i*i), strconv.Itoa(d))
	}

	return
}

File: code wars/7kyu/Count the Digit/main.ts
export class G964 {
    public static nbDig(n: number, d: number): number {
        let count = 0
        const arr: Array<number> = [...Array(n+1).keys()]
        arr.forEach(e => {
          const num:string = e**2 + ''
          count += num.split('').filter(e => e === d.toString()).length
        })
        return count
    }
}
// or
export class G964 {
    public static nbDig(n: number, d: number): number {
        let count: number = 0;
        for (let k: number = 0; k <= n; k++) {
            count += (k * k).toString().split(d.toString()).length - 1;
        }
        return count;
    }
}